lib/Core/formatNumberForLocale.js

```diff
@@ -32,10 +35,14 @@ function formatNumberForLocale(number, options) {
     options.useGrouping = defaultValue(options.useGrouping, false);
     options.maximumFractionDigits = defaultValue(options.maximumFractionDigits, 20);
     if (toLocaleStringSupportsOptions()) {
+        // The order of options may lead to equivalent keys being cached multiple times, but this is still better than creating a new Intl.NumberFormat every time.
```

Comment:
Can you change options into a sorted array of two-tuples and stringify that? Just a thought.